### PR TITLE
Username search and other minor bug fixes

### PR DESCRIPTION
**Pull request by Aryex82**
Issue: #483

Description: 
- Adds functionality to search based on single and multiple username.
- Fixes an issue where cards were duplicated when filter button was clicked multiple time in a short duration.
- Adds an error message element.
- Displays the extension cards with a spinner when user and task details are being fetched from the server. Previously all the cards where displayed only after all the data had been fetched.
- Adds minor css and animation changes.

Backend changes
- [] Yes
- [X] No

Frontend Changes
- [X] Yes
- [] No

Database changes
- [] Yes
- [X] No

Breaking changes
- [ ] Yes
- [X] No

Video after the change: 

https://github.com/Real-Dev-Squad/website-dashboard/assets/98796547/d03ac390-fb06-4326-b0d2-491d5aed2543

Test stats: 
Screenshot 2023-09-07 at 10 05 54 PM

### CODE REVIEW

**Review by kotesh-arya — APPROVED** [review]

File: extension-requests/local-utils.js [hunk 1]
```diff
@@ -130,6 +130,18 @@ async function getUserDetails(username) {
...
+    credentials: 'include',
+    method: 'GET',
+    headers: {
+      'Content-type': 'application/json',
+    },
+  });
+
+  return await res.json();
+}
+
```
Comment (fakhruddinkw) [inline extension-requests/local-utils.js:144]: As we are making request here, can we have a try catch here?

**Comment by iamitprakash** [comment]
@Aryex82  is this behind feature flag

**Review by iamitprakash — APPROVED** [review]

**Comment by iamitprakash** [comment]
@Aryex82 is this behind feature flag

**Comment by Aryex82** [comment]
> @Aryex82 is this behind feature flag

Yes.

**Review by gauravsinhaweb — APPROVED** [review]
LGTM

**Pull request merged**